app_frontend/src/components/Greeting.js

```diff
@@ -0,0 +1,35 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchGreeting } from '../actions/greetingActions';
+
+const Greeting = () => {
+  const dispatch = useDispatch();
+  const { greeting, loading, error } = useSelector((state) => state.greeting);
+
+  useEffect(() => {
+    dispatch(fetchGreeting());
+  }, [dispatch]);
+
+  return (
+    <div>
+      {loading && <p>Loading...</p>}
+      {error && (
+        <p>
+          Error:
+          {error}
+        </p>
+      )}
+      {greeting && (
+        <p>
```

Comment:
- [ ] Great job working on the backend side but I highly recommend  adding this in your javascript folder since if we want to connect our rails API we should add the frontend in the javascript folder found in the app itself instead of creating a folder contain the react app, you can check this [article](https://blog.appsignal.com/2022/08/03/connect-a-ruby-on-rails-app-with-react-in-a-monolith.html) to know more about how to integrate a react app with ruby on rails app  🙏🏻⭐🚀💯 since as mentioned in the requirements to add it in javascript folder as you learned in previous lessons 
![image](https://github.com/jcmunav63/rails-react-hello-world-1/assets/58553711/841561ba-2871-4ae0-9ff8-a4ae51903675)




![image](https://github.com/jcmunav63/rails-react-hello-world-1/assets/58553711/0f3531c8-a487-4bea-ab3d-1d3537defc6f)